File: _tutorials/tutorial-source-code/Tree/bintree.java
import bridges.connect.Bridges;
import bridges.base.BinTreeElement;


public class bintree {
	public static void main(String[] args) throws Exception{
    	
   					//create the Bridges object
		Bridges bridges = new Bridges(9, "YOUR_API_KEY", "YOUR_USER_ID");

		bridges.setTitle("A Huffman Coding Tree Example");
		bridges.setDescription("[Binary Tree Application]: Huffman tree coding the symbols C, D, E, K, L, M, U, Z with frequencies 32, 42, 120, 7, 42, 24, 37, 2");

   					//create  an   tree
		BinTreeElement&lt;Integer&gt; t0 = new BinTreeElement&lt;Integer&gt;("306", 0);
		BinTreeElement&lt;Integer&gt; t1 = new BinTreeElement&lt;Integer&gt;("120\\nE", 1);
		BinTreeElement&lt;Integer&gt; t2 = new BinTreeElement&lt;Integer&gt;("186", 2);
		BinTreeElement&lt;Integer&gt; t3 = new BinTreeElement&lt;Integer&gt;("79", 3);
		BinTreeElement&lt;Integer&gt; t4 = new BinTreeElement&lt;Integer&gt;("37\\nU", 4);
		BinTreeElement&lt;Integer&gt; t5 = new BinTreeElement&lt;Integer&gt;("42\\nD", 5);
		BinTreeElement&lt;Integer&gt; t6 = new BinTreeElement&lt;Integer&gt;("42\\nL", 6);
		BinTreeElement&lt;Integer&gt; t7 = new BinTreeElement&lt;Integer&gt;("107", 7);
		BinTreeElement&lt;Integer&gt; t8 = new BinTreeElement&lt;Integer&gt;("65", 8);
		BinTreeElement&lt;Integer&gt; t9 = new BinTreeElement&lt;Integer&gt;("32\\nC", 9);
		BinTreeElement&lt;Integer&gt; t10 = new BinTreeElement&lt;Integer&gt;("33", 10);
		BinTreeElement&lt;Integer&gt; t11 = new BinTreeElement&lt;Integer&gt;("9", 11);
		BinTreeElement&lt;Integer&gt; t12 = new BinTreeElement&lt;Integer&gt;("2\\nZ", 12);
		BinTreeElement&lt;Integer&gt; t13 = new BinTreeElement&lt;Integer&gt;("7\\nK", 13);
		BinTreeElement&lt;Integer&gt; t14 = new BinTreeElement&lt;Integer&gt;("24\\nM", 14);

		t0.setLeft(t1);
		t0.setRight(t2);
		t2.setLeft(t3);
		t2.setRight(t7);
		t3.setLeft(t4);
		t3.setRight(t5);
		t7.setLeft(t6);
		t7.setRight(t8);
		t8.setLeft(t9);
		t8.setRight(t10);
		t10.setLeft(t11);
		t10.setRight(t14);
		t11.setLeft(t12);
		t11.setRight(t13);

		t0.getVisualizer().setColor("red");
                                    // color the leaf nodes that represent the code letters
		t1.getVisualizer().setColor("orange");
		t4.getVisualizer().setColor("orange");
		t5.getVisualizer().setColor("orange");
		t6.getVisualizer().setColor("orange");
		t9.getVisualizer().setColor("orange");
		t12.getVisualizer().setColor("orange");
		t13.getVisualizer().setColor("orange");
		t14.getVisualizer().setColor("orange");
		
             					//set visualizer type
		bridges.setDataStructure(t0);
        						// visualize the tree
		bridges.visualize();
    }
}

File: tutorials/testing/python/bintree.py
from bridges.bridges import *
from bridges.bin_tree_element import *
import sys


def main():
    # create the Bridges object, set credentials
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")

    # title, description
    bridges.set_title("A Huffman Coding (Binary) Tree Example")
    bridges.set_description("[Binary Tree Application]: Huffman tree coding the symbols C, D, E, K, L, M, U, Z with frequencies 32, 42, 120, 7, 42, 24, 37, 2 "
            + "Root is set to red and leaf nodes are set to orange.")

    # build the tree
    t0 = BinTreeElement(label="306", e=0)
    t1 = BinTreeElement(label="120\\nE", e=1)
    t2 = BinTreeElement(label="186", e=2)
    t3 = BinTreeElement(label="79", e=3)
    t4 = BinTreeElement(label="37\\nU", e=4)
    t5 = BinTreeElement(label="42\\nD", e=5)
    t6 = BinTreeElement(label="42\\nL", e=6)
    t7 = BinTreeElement(label="107", e=7)
    t8 = BinTreeElement(label="65", e=8)
    t9 = BinTreeElement(label="32\\nC", e=9)
    t10 = BinTreeElement(label="33", e=10)
    t11 = BinTreeElement(label="9", e=11)
    t12 = BinTreeElement(label="2\\nZ", e=12)
    t13 = BinTreeElement(label="7\\nK", e=13)
    t14 = BinTreeElement(label="24\\nM", e=14)

    t0.left = t1
    t0.right = t2
    t2.left = t3
    t2.right = t7
    t3.left = t4
    t3.right = t5
    t7.left = t6
    t7.right = t8
    t8.left = t9
    t8.right = t10
    t10.left = t11
    t10.right = t14
    t11.left = t12
    t11.right = t13

    # color the leaf nodes to represent the code letters
    t0.color = "red"
    t1.color = "orange"
    t4.color = "orange"
    t5.color = "orange"
    t6.color = "orange"
    t9.color = "orange"
    t12.color = "orange"
    t13.color = "orange"
    t14.color = "orange"

    # set visualizer type
    bridges.set_data_structure(t0)
    # visualize the tree
    bridges.visualize()


if __name__ == "__main__":
    main()

